BackendApartmentReservation/BackendApartmentReservation/Infrastructure/Exceptions/GlobalExceptionFilter.cs

```diff
@@ -0,0 +1,32 @@
+﻿using BackendApartmentReservation.Extensions.Json;
+
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using NLog;
+
+namespace BackendApartmentReservation.Infrastructure.Exceptions
+{
+    public class GlobalExceptionFilter : IExceptionFilter
+    {
+        public void OnException(ExceptionContext context)
+        {
+            var logger = new LogFactory().GetCurrentClassLogger();
+            var exception = context.Exception;
+            logger.Error($"Uncaught exception happened. Error will be responded.\n\nException:{exception.ToJson()}");
+
+            context.ExceptionHandled = true;
+
+            var responseObject = new
+            {
+                ErrorCode = ErrorCodes.GenericInternalServerError,
+                Exception = "dev" == "dev" ? exception : null

```

Comment:
**Question**
What's the reason behind this check? It will always be true.

Reply:
This is a preparation for local/dev/prod environment I am planning to introduce with upcoming PRs